### ISSUE TITLE
One-of-many relationships don't use where condition in inner join

### ISSUE THREAD

**Issue by mnabialek** [laravel/framework#39131]
- Laravel Version: 8.60.0
- PHP Version: 8.0.11
- Database Driver & Version: MySQL 8.0.26

### Description:

It seems `oldestOfMany` (and probably also `latestOfMany`) introduced here https://github.com/laravel/framework/pull/37362/files completely in fact ignores any where conditions added to relationship although there is no clear mention in documentation that `where` conditions will be ignored

### Steps To Reproduce:

I have relationship defined like this:

```
    public function mainMediaPhoto()
    {
        return $this->hasOne(Media::class, 'product_id')
            ->where('type', Media::TYPE_PHOTO)
            ->oldestOfMany()
            ->withDefault(['type' => Media::TYPE_PHOTO]);
    }
```    

and it generates query like this:

```
SELECT
	*
FROM
	`media`
	INNER JOIN (
		SELECT
			MIN(`id`) AS `id`,
			`media`.`product_id`
		FROM
			`media`
		WHERE
			`media`.`product_id` in(9630, 9631)
		GROUP BY
			`media`.`product_id`) AS `mainMediaPhoto` ON `mainMediaPhoto`.`id` = `media`.`id`
	AND `mainMediaPhoto`.`product_id` = `media`.`product_id`
WHERE
	`type` = 'photo';
```

Although there is `where `type` = 'photo'` condition put in SQL query it was put outside of inner join which makes query working not as expected as it  first selects wrong records and then filter them.

Expected query should look in this case like this:

```
SELECT
	*
FROM
	`media`
	INNER JOIN (
		SELECT
			MIN(`id`) AS `id`,
			`media`.`product_id`
		FROM
			`media`
		WHERE
			`media`.`product_id` in(9630, 9631)
			AND `media`.`type` = 'photo'
		GROUP BY
			`media`.`product_id`) AS `mainMediaPhoto` ON `mainMediaPhoto`.`id` = `media`.`id`
	AND `mainMediaPhoto`.`product_id` = `media`.`product_id`
```

In documentation there is description about `oneOfMany`  here: https://laravel.com/docs/8.x/eloquent-relationships#advanced-has-one-of-many-relationships but it describes at first glance a bit more complicated scenario but in fact it could be used here to correct query. If I define relationship like this:

```
    public function mainMediaPhoto()
    {
        return $this->hasOne(Media::class, 'product_id')
            ->ofMany(['id' => 'MIN'], function($query) {
                $query->where('type', Media::TYPE_PHOTO);
            })
            ->withDefault(['type' => Media::TYPE_PHOTO]);
    }
 ```

then results are as expected but it doesn't seem to be obvious that completely different syntax has to be used to make where condition working fine.

**Comment by driesvints** [comment]
@cbl thanks! Is there any way we can make this all more clear in the docs?

@mnabialek please see the above.

**Comment by driesvints** [comment]
@cbl yeah I think we need a specific example of this since we've already gotten quite a bit of questions about it. Could you maybe PR that to the docs?
